File: frontend/src/interface/zone.ts
export interface ZoneInterface {
  ID?: number;
  zone_name?: string;
  zone_price?: number | string;
  zonetype_id?: number;
  zone_type?: string;
  seat_available?: Array<{ seatavailable_status?: string | null }> | null;
  showdate_id?: number;
  venue_id?: number;
  capacity?: number;
  pending_holds?: number;
  seat_sold?: number;
  availableSeats?: number;
  available_count?: number;
}

File: backend/entity/zone.go
package entity

import (
	"gorm.io/gorm"
)
type Zone struct {
	gorm.Model
	ShowDateID uint `json:"showdate_id"`
	ShowDate *ShowDate `gorm:"foreignKey:ShowDateID" json:"show_date"`
	VenueID uint `json:"venue_id"`
	Venue *Venue `gorm:"foreignKey:VenueID" json:"venue"`
	ZoneName string `json:"zone_name"`
	ZoneTypeID uint `json:"zonetype_id"`
	ZoneType *ZoneType `gorm:"foreignKey:ZoneTypeID" json:"zone_type"`
	ZonePrice float32 `json:"zone_price"`
	Capacity int `json:"capacity"`
	Seats      []SeatAvailable `gorm:"foreignKey:ZoneID" json:"seat_available"`
}